lib/std/ascii.zig

```diff
@@ -267,18 +267,24 @@ test "ASCII character classes" {
 
 /// Writes a lower case copy of `ascii_string` to `output`.
 /// Asserts `output.len >= ascii_string.len`.
+/// `output` and `ascii_string` may alias.
```

Comment:
The intention is probably to mean "exact aliasing", i.e. without offset.
`output = ascii_string[1..]` would corrupt the input with the output (just like in `std.mem.copyForwards`).
From there, the more accurate phrasing would be something like
"If `output` and `ascii_string` overlap, output.ptr must be <= ascii_string.ptr."
(same for `upperString` below)

Reply:
Thanks for the catch, I did mean exact aliasing. I updated the comment to be more specific. In the case of "output.ptr must be <= ascii_string.ptr", I do not think it would make sense to support that as in the future that could be problematic for an SIMD implementation of the function.